### PR TITLE
Move instances of Client.createPost() in components to an action

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Move instances of Client.createPost() in components to an action

#### Ticket Link
GitHub: https://github.com/mattermost/platform/issues/4589

### CODE REVIEW

File: webapp/actions/post_actions.jsx [hunk 1]
```diff
@@ -252,3 +252,37 @@ export function loadProfilesForPosts(posts) {
...
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECEIVED_POST,
+                post: data
+            });
+        },
+
+        (err) => {
+            if (err.id === 'api.post.create_post.root_id.app_error') {
+                // this should never actually happen since you can't reply from this textbox
+                this.showPostDeletedModal();
```
Comment (enahum) [inline webapp/actions/post_actions.jsx]: This should give an error as `showPostDeleteModal` is a function part of the component

File: webapp/actions/post_actions.jsx [hunk 2]
```diff
@@ -252,3 +252,37 @@ export function loadProfilesForPosts(posts) {
...
+
+                PostStore.removePendingPost(post.pending_post_id);
+            } else {
+                post.state = Constants.POST_FAILED;
+                PostStore.updatePendingPost(post);
+                this.forceUpdate();
```
Comment (enahum) [inline webapp/actions/post_actions.jsx]: This is not a react component, am I right?

File: webapp/actions/post_actions.jsx [hunk 3]
```diff
@@ -252,3 +252,37 @@ export function loadProfilesForPosts(posts) {
...
+            }
+
+            this.setState({
```
Comment (enahum) [inline webapp/actions/post_actions.jsx]: again you cannot set state here

File: webapp/components/create_comment.jsx [hunk 4]
```diff
@@ -135,31 +135,8 @@ export default class CreateComment extends React.Component {
...
-                    PostStore.updatePendingPost(post);
-                }
-
-                this.setState({
-                    submitting: false
-                });
-            }
-        );
+        var doLoadPost = false;
+        createPost(post, doLoadPost);
```
Comment (enahum) [inline webapp/components/create_comment.jsx]: Why not just pass the bool to the call? as in `createPost(post, false);`

File: webapp/actions/post_actions.jsx (hunk 1 above)
Comment (cpanato) [inline webapp/actions/post_actions.jsx]: ok, will move to the original part

File: webapp/actions/post_actions.jsx (hunk 2 above)
Comment (cpanato) [inline webapp/actions/post_actions.jsx]: not sure

File: webapp/actions/post_actions.jsx (hunk 3 above)
Comment (cpanato) [inline webapp/actions/post_actions.jsx]: ok

File: webapp/components/create_comment.jsx (hunk 4 above)
Comment (cpanato) [inline webapp/components/create_comment.jsx]: ok

File: webapp/actions/post_actions.jsx (hunk 1 above)
Comment (cpanato) [inline webapp/actions/post_actions.jsx]: I might need some help here to understand where I need to add the showPostDeletedModal(); @enahum

File: webapp/actions/post_actions.jsx [hunk 5]
```diff
@@ -252,3 +252,30 @@ export function loadProfilesForPosts(posts) {
 
     AsyncClient.getProfilesByIds(list);
 }
+
+export function createPost(post, doLoadPost) {
```
Comment (enahum) [inline webapp/actions/post_actions.jsx]: the only thing im seeing is in createPost you need to add the two callbacks

Comment (cpanato) [inline webapp/actions/post_actions.jsx]: the success and error?

**Comment by jwilander** [comment]
Nice work @cpanato! Can you give this a rebase and then I can merge it in?

File: webapp/actions/post_actions.jsx [hunk 6]
```diff
@@ -272,3 +272,38 @@ export function removeReaction(channelId, postId, emojiName) {
...
+            });
+
+            if (success) {
+                success(data);
+            }
+        },
+
+        (err) => {
+            if (err.id === 'api.post.create_post.root_id.app_error') {
+                // this should never actually happen since you can't reply from this textbox
```
Comment (jwilander) [inline webapp/actions/post_actions.jsx]: Sorry @cpanato, just one more thing. Can you remove this comment? I don't think it makes sense in this context

File: webapp/components/create_comment.jsx [hunk 7]
```diff
@@ -163,31 +162,36 @@ export default class CreateComment extends React.Component {
...
-                    post: data
-                });
+
+        PostActions.createPost(post, false,
+            () => {
+                // DO nothing.
             },
             (err) => {
                 if (err.id === 'api.post.create_post.root_id.app_error') {
+                    // this should never actually happen since you can't reply from this textbox
```
Comment (jwilander) [inline webapp/components/create_comment.jsx]: Can you remove it from here as well?

File: webapp/components/post_view/components/pending_post_options.jsx [hunk 8]
```diff
@@ -26,25 +22,21 @@ export default class PendingPostOptions extends React.Component {
...
-                member.msg_count = channel.total_msg_count;
-                member.last_viewed_at = (new Date()).getTime();
-                ChannelStore.storeMyChannelMember(member);
+        createPost(post, true,
+            () => {
+                // DO nothing.
+            },
+            (err) => {
+                if (err.id === 'api.post.create_post.root_id.app_error') {
+                    // this should never actually happen since you can't reply from this textbox
```
Comment (jwilander) [inline webapp/components/post_view/components/pending_post_options.jsx]: And here too?

**Pull request merged**